benchmarks/bench/merge_sort/pairs.cu

```diff
@@ -0,0 +1,141 @@
+#include <cub/device/device_merge_sort.cuh>
+
+#include <nvbench_helper.cuh>
+
+// %RANGE% TUNE_TRANSPOSE trp 0:1:1
+// %RANGE% TUNE_LOAD ld 0:2:1
+// %RANGE% TUNE_ITEMS_PER_THREAD ipt 7:24:1
+// %RANGE% TUNE_THREADS_PER_BLOCK_POW2 tpb 6:10:1
+
+#ifndef TUNE_BASE
+#define TUNE_THREADS_PER_BLOCK (1 << TUNE_THREADS_PER_BLOCK_POW2)
+#endif
+
+#if !TUNE_BASE
+#if TUNE_TRANSPOSE == 0
+#define TUNE_LOAD_ALGORITHM cub::BLOCK_LOAD_DIRECT
+#define TUNE_STORE_ALGORITHM cub::BLOCK_STORE_DIRECT
+#else
+#define TUNE_LOAD_ALGORITHM cub::BLOCK_LOAD_WARP_TRANSPOSE
+#define TUNE_STORE_ALGORITHM cub::BLOCK_STORE_WARP_TRANSPOSE
+#endif
+
+#if TUNE_LOAD == 0
+#define TUNE_LOAD_MODIFIER cub::LOAD_DEFAULT
+#elif TUNE_LOAD == 1
+#define TUNE_LOAD_MODIFIER cub::LOAD_LDG
+#else
+#define TUNE_LOAD_MODIFIER cub::LOAD_CA
+#endif
+
+template <typename KeyT>
+struct policy_hub_t
+{
+  struct policy_t : cub::ChainedPolicy<300, policy_t, policy_t>
+  {
+    using MergeSortPolicy =
+      cub::AgentMergeSortPolicy<TUNE_THREADS_PER_BLOCK,
+                                cub::Nominal4BItemsToItems<KeyT>(TUNE_ITEMS_PER_THREAD),
+                                TUNE_LOAD_ALGORITHM,
+                                TUNE_LOAD_MODIFIER,
+                                TUNE_STORE_ALGORITHM>;
+  };
+
+  using MaxPolicy = policy_t;
+};
+#endif
+
+template <typename KeyT, typename ValueT, typename OffsetT>
+void merge_sort_keys(nvbench::state &state, nvbench::type_list<KeyT, ValueT, OffsetT>)
+{
+  using key_t            = KeyT;
+  using value_t          = ValueT;
+  using key_input_it_t   = key_t *;
+  using value_input_it_t = value_t *;
+  using key_it_t         = key_t *;
+  using value_it_t       = value_t *;
+  using offset_t         = OffsetT;
+  using compare_op_t     = less_t;
+
+#if !TUNE_BASE
+  using policy_t   = policy_hub_t<key_t>;
+  using dispatch_t = cub::DispatchMergeSort<key_input_it_t,
+                                            value_input_it_t,
+                                            key_it_t,
+                                            value_it_t,
+                                            offset_t,
+                                            compare_op_t,
+                                            policy_t>;
+#else
+  using dispatch_t = cub::
+    DispatchMergeSort<key_input_it_t, value_input_it_t, key_it_t, value_it_t, offset_t, compare_op_t>;
+#endif
+
+  // Retrieve axis parameters
+  const auto elements       = static_cast<std::size_t>(state.get_int64("Elements{io}"));
+  const bit_entropy entropy = str_to_entropy(state.get_string("Entropy"));
+
+  thrust::device_vector<key_t> keys_buffer_1(elements);
+  thrust::device_vector<key_t> keys_buffer_2(elements);
+  thrust::device_vector<value_t> values_buffer_1(elements);
+  thrust::device_vector<value_t> values_buffer_2(elements);
+
+  gen(seed_t{}, keys_buffer_1);
+
+  key_t *d_keys_buffer_1   = thrust::raw_pointer_cast(keys_buffer_1.data());
```

Comment:
Not sure if you prefer the `raw_pointer_cast`, but you can also do it like this which is more concise. 
```suggestion
  key_t *d_keys_buffer_1   = keys_buffer_1.data().get();
```